crates/accelerate/src/sparse_observable.rs

```diff
@@ -327,11 +201,8 @@ pub enum CoherenceError {
     DuplicateIndices,
     #[error("the provided qubit mapping does not account for all contained qubits")]
     IndexMapTooSmall,
-}
-impl From<CoherenceError> for PyErr {
-    fn from(value: CoherenceError) -> PyErr {
-        PyValueError::new_err(value.to_string())
-    }
+    #[error("cannot shrink the qubit count in an observable from {current} to {target}")]
+    NotEnoughQubits { current: usize, target: usize },
```

Comment:
I think this is just a straight-up code change?  Might be better to do that in a separate PR, if you think it's necessary.

Reply:
Ok sure, if it's actually an important part of the split, it's fine to do here. (Part of the difficulty in reviewing this kind of PR!)